### PR TITLE
Additional: max size and number of files

### PR DESCRIPTION
**Pull request by RobJohnston**
This is a continuation of abandoned PR https://github.com/jquery-validation/jquery-validation/pull/1512, in that it validates the maximum size of a selected file.

Since the "file" type can contain the multiple attribute, another method was added to validate the maximum size of all selected files.  Finally, a third method was added to validate the maximum number of files.

### CODE REVIEW

File: src/additional/maxfiles.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//Limit the number of files in a FileList.
+$.validator.addMethod( "maxfiles", function( value, element, param ) {
+	if ( this.optional( element ) ) {
+		return true;
+	}
+
+	if ( $( element ).attr( "type" ) === "file" ) {
+		if ( element.files && element.files.length ) {
+			if ( element.files.length > param ) {
```
Comment (Arkni) [inline src/additional/maxfiles.js]: Can you please merge the above two `if` blocks?

File: src/additional/maxfiles.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//Limit the number of files in a FileList.
```
Comment (Arkni) [inline src/additional/maxfiles.js]: Just for consistency with the other comments in this project, can you add a space before the comment?
The same goes for other comments bellow.

File: src/additional/maxfiles.js (hunk 1 above)
Comment (RobJohnston) [reply src/additional/maxfiles.js]: Which two?  There are three of them.  Lines 8 and 9, I presume?

Comment (Arkni) [reply src/additional/maxfiles.js]: Yes, 8 and 9.

**Comment by RobJohnston** [comment]
@Arkni  Done.

**Comment by Arkni** [comment]
Thanks a lot for your contribution :)

**Pull request merged**